src/sagas/projects.js

```diff
@@ -87,6 +92,36 @@ export function* updateProjectSource() {
   yield* saveCurrentProject();
 }
 
+export async function importBeautify() {
+  return retryingFailedImports(
+    () => import(
+      /* webpackChunkName: "mainAsync" */
+      'js-beautify',
+    ),
+  );
+}
+
+export function* loadAndBeautifyProjectSource() {
+  const beautify = yield call(importBeautify);
+
+  const state = yield select();
+  const currentProject = getCurrentProject(state);
+
+  const beautifiedSourcesMap = new Map();
+  for (const language of Reflect.ownKeys(currentProject.sources)) {
+    const source = currentProject.sources[language];
+    const newSource = format(
```

Comment:
oh i'm reading more into sagas, i think i should actually be forking here, hold on